pytorch_lightning/trainer/data_loading.py

```diff
@@ -271,7 +271,7 @@ def _reset_eval_dataloader(
         """
         # always get the loaders first so we can count how many there are
         loader_name = f'{mode}_dataloader'
-        dataloaders = self.request_dataloader(getattr(model, loader_name))
+        dataloaders = self.request_dataloader(model, mode)
```

Comment:
not maybe in this PR, but maybe have the mode ad our LightEnum

Reply:
Yes, you are right. I will do this in another PR.